servers/src/common/adapters.rs

```diff
@@ -264,9 +264,36 @@ impl p2p::ChainAdapter for NetToChainAdapter {
 		}
 
 		// headers will just set us backward if even the last is unknown
-		let last_h = bhs.last().unwrap().hash();
+		let bhs_last = bhs.last().unwrap().clone();
+		let last_h = bhs_last.hash();
 		if let Ok(_) = w(&self.chain).get_block_header(&last_h) {
-			info!(LOGGER, "All known, ignoring");
+			if bhs.len() < p2p::MAX_BLOCK_HEADERS as usize - 1 {
+				info!(LOGGER, "All known, ignoring");
+				return true;
+			}
+
+			info!(
+				LOGGER,
+				"All known, ignoring. Update header_head and sync_head to {} at {}",
+				last_h,
+				bhs_last.height,
+			);
+
+			// Update header_head and sync_head to the last header of this Headers package
```

Comment:
> We only want to update the sync_head, to allow us to build a new locator.

Agree. Update header_head was your proposal (in issue #1612) 😄 

> Maybe worth considering just simply re-processing all the headers and letting the existing logic handle this?

I would like to keep this "All known, ignoring" behavior, since it's an obvious exception condition and easy to read this. Ignoring is much faster than re-processing.